### DISCUSSION TITLE
Shap Artifact: How do I use the explainer artifact created by mlflow.evaluate?

### DISCUSSION THREAD

**Discussion by khojarohan4** [mlflow/mlflow#6454] [Q&A] [answered]
I trained a hist gradient boosting model. I used the mlflow.evaluate module along with some custom metrics per my need. All is fine and now I have 2 artifacts folder namely explainer and model. The model folder I understand where the model is pickled and I can use it for prediction. How do I use the explainer.shap file ? Help and support is appreciated.

The explainer used is kernel type

I have tried something like this

ex2 = joblib.load(filename='explainer.shap')
shap_values2 = ex2.shap_values(X_test.drop('target', axis=1).iloc[0,:])

It gives me error:

shap_values() missing 1 required positional argument: 'X'

**Comment by WeichenXu123** [comment, marked as answer]
I think you should use `mlflow.shap.load_explainer` to load the saved explainer.
After you get the loaded `explainer` object, then you can call `explainer.shap_values(X)` (X is the sampled dataset you want to explain)

**Reply by khojarohan4** [reply]
This is now solved. I have tried the below and I get the shap values. Thank you @WeichenXu123 
I can now use this to plot the charts as per the requirements
ex = mlflow.shap.load_explainer("<path>/explainer_on_data_churn")
ex.shap_values(X_test.iloc[0,:])
